src/transformers/models/auto/auto_factory.py

```diff
@@ -502,8 +502,9 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
                 if token is not None:
                     adapter_kwargs["token"] = token
 
+            revision = adapter_kwargs.pop("revision", hub_kwargs.get("revision"))
```

Comment:
```suggestion
            revision = adapter_kwargs.get("revision", hub_kwargs.get("revision"))
```
Since `adapter_kwargs` gets passed later in `from_pretrained` let's use `get`